card/src/test/java/com/adyen/checkout/card/internal/util/CardValidationUtilsTest.kt

```diff
@@ -8,168 +8,520 @@
-            enableLuhnCheck = false,
-            isBrandSupported = true
-        )
-        assertEquals(CardNumberValidation.VALID, validation)
-    }
+        @Test
+        fun `luhn check fails then result should be invalid`() {
+            val number = "8475 1789 7235 6236"
+            val validation = CardValidationUtils.validateCardNumber(
+                number = number,
+                enableLuhnCheck = true,
+                isBrandSupported = true
+            )
+            assertEquals(CardNumberValidation.INVALID_LUHN_CHECK, validation)
+        }
 
-    @Test
-    fun isInMaxYearRange_ExpiryDateThirtyYearsInFuture_ExpectTrue() {
-        val mockCalendarInstance = GregorianCalendar(2022, 4, 23) // 23/05/2022
-        val expiryDate = ExpiryDate(12, 2052) // 12/2052 (last valid date in future)
-        val actual = CardValidationUtils.isInMaxYearRange(expiryDate, mockCalendarInstance)
-        assertEquals(true, actual)
+        @Test
+        fun `luhn check fails but luhn check is disabled then result should be valid`() {
+            val number = "192382023091310912"
+            val validation = CardValidationUtils.validateCardNumber(
+                number = number,
+                enableLuhnCheck = false,
+                isBrandSupported = true
+            )
+            assertEquals(CardNumberValidation.VALID, validation)
+        }
     }
 
-    @Test
-    fun isInMaxYearRange_ExpiryDateMoreThanThirtyYearsInFuture_ExpectFalse() {
-        val mockCalendarInstance = GregorianCalendar(2022, 4, 23) // 23/05/2022
-        val expiryDate = ExpiryDate(1, 2053) // 01/2053 (first invalid date in future)
-        val actual = CardValidationUtils.isInMaxYearRange(expiryDate, mockCalendarInstance)
-        assertEquals(false, actual)
-    }
+    @Nested
+    @DisplayName("when validating expiry date and")
+    inner class ValidateExpiryDateTest {
 
-    @Test
-    fun isInMaxYearRange_ExpiryDateEightYearsInFuture_ExpectTrue() {
-        val mockCalendarInstance = GregorianCalendar(2022, 4, 23) // 23/05/2022
-        val expiryDate = ExpiryDate(1, 2030) // 01/2030
-        val actual = CardValidationUtils.isInMaxYearRange(expiryDate, mockCalendarInstance)
-        assertEquals(true, actual)
-    }
+        @Test
+        fun `date is 30 years in the future then result should be valid`() {
+            val mockCalendarInstance = GregorianCalendar(2022, 4, 23) // 23/05/2022
+            val expiryDate = ExpiryDate(12, 2052) // 12/2052 (last valid date in future)
+            val actual = CardValidationUtils.validateExpiryDate(
+                expiryDate = expiryDate,
+                fieldPolicy = Brand.FieldPolicy.REQUIRED,
+                calendar = mockCalendarInstance
+            )
+            assertEquals(FieldState(expiryDate, Validation.Valid), actual)
+        }
+
+        @Test
+        fun `date is more than 30 years in the future then result should be invalid as not within max range`() {
```

Comment:
A parameterised test would have saved you some work and is also easier to maintain

Reply:
Considered it multiple times but ended up not doing it, do you think it would be valuable to change the current tests to parameterised tests?

Reply:
No not necessarily. I do prefer them though as it saves me from writing extra code and it's easier to add/change cases later on